File: C - 2D Arrays/Solution.swift
var arr = Array(repeating: Array( repeating: 0, count: 6), count: 6)

for i in 0 ... 5 {
    let tmp = readLine()!.components(separatedBy: " ").map{ Int($0)! }
    arr[i] = tmp.map {
        Int($0)
    }
}

var max = -9 * 7;

for i in 0 ... 5 {
    for j in 0 ... 5 {
        if j + 2 < 6 && i + 2 < 6 {
            let top = arr[i][j] + arr[i][j + 1] + arr[i][j + 2]
            let mid = arr[i + 1][j + 1]
            let bot = arr[i + 2][j] + arr[i + 2][j + 1] + arr[i + 2][j + 2];
            let sum = top + mid + bot
            
            if sum > max {
                max = sum
            }
        }
    }
}

print(max)

File: C - 2D Arrays/Solution.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<vector<int>> arr(6, vector<int>(6));
    for (int i = 0; i < 6; i++) {
        for (int j = 0; j < 6; j++) {
            cin >> arr[i][j];
        }
    }

    int max = -9 * 7;
    for (int i = 0; i < 6; i++) {
        for (int j = 0; j < 6; j++) {
            if (j + 2 < 6 && i + 2 < 6) {
                int sum = arr[i][j] + arr[i][j + 1] + arr[i][j + 2] + arr[i + 1][j + 1] + arr[i + 2][j] + arr[i + 2][j + 1] + arr[i + 2][j + 2];
                if (sum > max) max = sum;
            }
        }
    }

    cout << max;

    return 0;
}
